src/redux/actions/samples/updateSample.js

```diff
@@ -3,34 +3,97 @@ import _ from 'lodash';
 
 import endUserMessages from 'utils/endUserMessages';
 
```

Comment:
I feel that this file it's too messy due to having to support both v1 & v2. It's not clear which actions are dispatched and when for each version (as evidenced by Iva's comment) and I can totally see people inadvertently adding duplicated notifications. I have a hard time telling whether the notifications are correct or not. I am not sure what's the best course of action here. The idea is to make it as easy as possible to transition from v1 to v2. However, in this PR it will take a bit more than just removing the extra v1 code. For example, you added an extra function `sendApiV2Request`. But when we remove v1, will be a nuisance because the handleError is duplicated unnecessarily because the try/catch is too large and includes both versions. I would probably do either:
1. add a different function for v1 and v2 and add EVERYTHING inside each (that is, all the try/catch, dispatches, etc... so that when we migrate we can just complete remove v1 without any more changes. Somethine like:
```
if (config.currentApiVersion === api.V1) {
     updateSamplesV1(...)
else {
    updateSamplesV2(...)
}
```

The the other option would be to share all the flow (if that's the desired behavior which I'm not sure anymore) and just change the actual dispatch
```
const updateSample = (
  sampleUuid,
  diff,
) => async (dispatch, getState) => {
  const sample = _.cloneDeep(getState().samples[sampleUuid]);
    if (config.currentApiVersion === api.V1) {

      await dispatch(saveSamples(sample.projectUuid, newSample, true, true, false));
   } else {
      await whatever for v2
   }
      dispatch({
        type: SAMPLES_UPDATE,
        payload: {
          sampleUuid,
          sample: diff,
        },
 try {

  } catch (e) {
    const errorMessage =  handleError(e, endUserMessages.ERROR_SAVING);
    dispatch({
      type: SAMPLES_ERROR,
      payload: {
        error: errorMessage,
      },
    });
  }
```

imo something like first approach is better for the reasons listed above. I think you are trying to mix both approaches (making a function for v2 but no the other, sharing some error handling but not all of it, etc...) which. just makes everything more confusing than needed. I'd prefer code duplication if you don't like any of these options

Reply:
This action is particularly complex because I needed to add the actions from `saveSamples` in here, I duplicated `handleError` because for api v1 it is duplicated as well, since `saveSample` does it once and then `updateSample` does it too. 

My objective was to make redux work in the same way as it works with v1, in case not doing that introduces any problems or unpredictable behavior.

Reply:
> 1. add a different function for v1 and v2 and add EVERYTHING inside each (that is, all the try/catch, dispatches, etc... > so that when we migrate we can just complete remove v1 without any more changes. Somethine like:

This was my initial implementation when I started working on adding api v2 endpoints but it was suggested in a code review that instead I try to keep the diverging code down.

Reply:
As I said before, in this case it is more messy than normal because I needed to add everything redux-wise that `saveSample` does, the duplicated error handling that I added was not by mistake but because of `saveSamples`, maybe a call to explain this in more detail might help so we can decide how to make this easier to follow.

Reply:
Resolved in a call, we decided to keep code for both actions (v1 and v2) in separate functions in this case becuase it makes things too messy otherwise.